modules/core/src/test/java/org/apache/ignite/cdc/CdcManagerTest.java

```diff
@@ -0,0 +1,429 @@
+
+        rollSegment.run();
+
+        for (int i = 0; i < 10_000; i++)
+            ign.cache(DEFAULT_CACHE_NAME).put(i, i);
+
+        File walDir = walDir(ign);
+
+        stopGrid(0);
+
+        List<File> segs = Arrays.stream(walDir.listFiles()).sorted()
+            .limit(2)
+            .collect(Collectors.toList());
+
+        int firstSegActLen = writtenLength(segs.get(0));
+
+        ByteBuffer walBuf = ByteBuffer.wrap(Files.readAllBytes(segs.get(0).toPath()));
+        ByteBuffer cdcBuf = ByteBuffer.wrap(cdcMgr.buf.array());
+        walBuf.limit(firstSegActLen);
+        cdcBuf.limit(firstSegActLen);
+
+        assertEquals(0, walBuf.compareTo(cdcBuf));
+
+        walBuf = ByteBuffer.wrap(Files.readAllBytes(segs.get(1).toPath()));
+        cdcBuf = ByteBuffer.wrap(cdcMgr.buf.array());
+        cdcBuf.position(firstSegActLen + 1);
+        cdcBuf.limit(firstSegActLen + 1 + WAL_SEG_SIZE);
+
+        assertEquals(0, walBuf.compareTo(cdcBuf));
+    }
+
+    /** Get WAL directory. */
+    private File walDir(IgniteEx ign) throws Exception {
+        return new File(
+            U.resolveWorkDirectory(
+                ign.configuration().getWorkDirectory(),
+                DataStorageConfiguration.DFLT_WAL_PATH,
+                false),
+            ign.context().pdsFolderResolver().resolveFolders().folderName());
+    }
+
+    /** @return Length of the all written records in the specified segment. */
+    private int writtenLength(File walSegment) throws Exception {
+        IgniteWalIteratorFactory factory = new IgniteWalIteratorFactory(log);
+
+        IgniteWalIteratorFactory.IteratorParametersBuilder params = new IgniteWalIteratorFactory.IteratorParametersBuilder()
+            .filesOrDirs(walSegment);
+
+        try (WALIterator walIt = factory.iterator(params)) {
+            while (walIt.hasNext())
+                walIt.next();
+
+            return walIt.lastRead().get().next().fileOffset();
+        }
+    }
+
+    /** */
+    private static class CdcManagerPluginProvider extends AbstractTestPluginProvider {
+        /** {@inheritDoc} */
+        @Override public String name() {
+            return "CdcManagerPluginProvider";
+        }
+
+        /** {@inheritDoc} */
+        @Override public <T> @Nullable T createComponent(PluginContext ctx, Class<T> cls) {
+            if (CdcManager.class.equals(cls))
+                return (T)new TestCdcManager();
+
+            return null;
+        }
+    }
+
+    /** Test {@link CdcManager} for testing WAL fsync guarantees. */
+    protected static class TestCdcManager extends GridCacheSharedManagerAdapter implements CdcManager {
+        /** Buffer to store collected data. */
+        private final ByteBuffer buf;
+
+        /** */
+        TestCdcManager() {
+            buf = ByteBuffer.allocate(2 * WAL_SEG_SIZE);
+
+            Arrays.fill(buf.array(), (byte)0);
+
+            buf.position(0);
+        }
+
+        /** */
+        public boolean stopped() {
```

Comment:
Can we use, `enabled` method with the same implementation instead of explicit `stopped` method?
```
        /** {@inheritDoc} */
        @Override public boolean enabled() {
            return isStopping();
        }
```

Reply:
No, it affects test. I just replace `stopped()` method with a test-local var